File: manual/data/cpp/tests/test_itertools_repeat.cpp
#include <gtest/gtest.h>
#include "../src/function_itertools_repeat.cpp"

class TestRepeatExample : public testing::Test {
protected:
    RepeatExample calc;
};

TEST_F(TestRepeatExample, RepeatInteger) {
    std::string result = calc.repeat_element(5, 3);
    EXPECT_EQ(result, "5, 5, 5");
}

TEST_F(TestRepeatExample, RepeatString) {
    std::string result = calc.repeat_element("apple", 2);
    EXPECT_EQ(result, "apple, apple");
}

TEST_F(TestRepeatExample, RepeatZeroTimes) {
    std::string result = calc.repeat_element(10, 0);
    EXPECT_EQ(result, "");
}

TEST_F(TestRepeatExample, NonNumeric) {
    std::string result = calc.repeat_element("banana", 4);
    EXPECT_EQ(result, "banana, banana, banana, banana");
}

TEST_F(TestRepeatExample, InvalidInput) {
    EXPECT_THROW({
        calc.repeat_element("apple", boost::any("two"));
    }, std::invalid_argument);
}
int main(int argc, char **argv) {
    testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: manual/data/python/tests/test_itertools_repeat.py
import unittest
from function_itertools_repeat import RepeatExample

class TestRepeatExample(unittest.TestCase):

    def test_repeat_integer(self):
        calc = RepeatExample()
        result = calc.repeat_element(5, 3)
        self.assertEqual(result, "5, 5, 5")

    def test_repeat_string(self):
        calc = RepeatExample()
        result = calc.repeat_element("apple", 2)
        self.assertEqual(result, "apple, apple")

    def test_repeat_zero_times(self):
        calc = RepeatExample()
        result = calc.repeat_element(10, 0)
        self.assertEqual(result, "")

    def test_non_numeric(self):
        calc = RepeatExample()
        result = calc.repeat_element("banana", 4)
        self.assertIsInstance(result, str)

    def test_invalid_input(self):
        calc = RepeatExample()
        with self.assertRaises(TypeError):
            calc.repeat_element("apple", "two")

if __name__ == '__main__':
    unittest.main()
